### ISSUE TITLE
calls to asm code always consider as indirect calls

### ISSUE THREAD

**Issue by bernardnongpoh** [SVF-tools/SVF#874]
Hi @yuleisui, I am working on a code where there are a lot of calls to asm code. All these calls are consider as icall in SVF with Andersen analysis. I would to clarify if you have specially handled  `call asm`

**Comment by yuleisui** [comment]
Could you show an example (e.g., c and llvm IR) for me understand a bit more?

**Comment by bernardnongpoh** [comment]
This is one of the asm instruction
`call asm sideeffect "","~{memory}"`

Thank you for your prompt reply as always

**Comment by yuleisui** [comment]
I am unsure it can be handled by SVF as icall. Why did you say svf treats asm as icall?

**Comment by bernardnongpoh** [comment]
Why did you say svf treats asm as icall?
Oh yes you're right!. LLVM treats these as icalls (Not SVF). Do you think we can safely ignore these icalls?

**Comment by yuleisui** [comment]
Is this callsite in `indCallSiteToFunPtrMap`?

**Comment by bernardnongpoh** [comment]
> Is this callsite in indCallSiteToFunPtrMap?
No. These are icalls that are not resolved by SVF (Andersen). `hasIndCSCallees(call)` return false.

**Comment by yuleisui** [comment]
Do you mean `icall` not indirect call?

**Comment by bernardnongpoh** [comment]
I mean idirect call

**Comment by yuleisui** [comment]
SVF's `indCallSiteToFunPtrMap` stores all indirect callsites. I believe `call asm sideeffect "","~{memory}"` must not be an indirect call?

**Comment by bernardnongpoh** [comment]
The callsites are coming from `andersen->getPAG()->getIndirectCallsites()`. `call asm sideeffect "","~{memory}"` are from this set.

**Comment by bernardnongpoh** [comment]
Yes, but according to my analysis. I am not seeing these calls are pointing to any functions. They are inserting `asm code` and not calling any functions.

**Comment by yuleisui** [comment]
I see. It could be related to memory access, writing or reading.

**Comment by bernardnongpoh** [comment]
One nice thing if we can do is to insert a custom node that point to asm code pointing by the `call asm`. The final callgraph can have these `call asm` as leaf node. In this way, we still retain the info and let users know about these `asm code`. Any thought on this would be highly appreciated.

**Comment by bernardnongpoh** [comment]
> I see. It could be related to memory access, writing or reading.

Yes might be!
